via/views/route_by_content/view.py

```diff
@@ -21,8 +21,12 @@ def route_by_content(context, request):
 
         return exc.HTTPFound(redirect_url, headers=_caching_headers(max_age=300))
 
-    url = HTMLRewriter.from_request(request).url_for(request.params)
-    headers = _cache_headers_for_http(status_code)
+    url, cacheable = HTMLRewriter.from_request(request).url_for(request.params)
+    headers = (
+        _cache_headers_for_http(status_code)
+        if cacheable
+        else {"Cache-Control": "no-cache"}
```

Comment:
There isn't actually any test coverage of this behavior. I want to remove it soon, so I thought I'd be lazy